server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -142,8 +146,8 @@ public ActivationKeyData updateActivationKey(
             serviceLevelValidator.validate(toUpdate.getOwner(), serviceLevel);
             toUpdate.setServiceLevel(serviceLevel);
         }
-        if (update.getReleaseVersion() != null) {
```

Comment:
I agree that it needs to go away, since the model representation of the release field on ActivationKey is just a String (which is how it's persisted in the db). The only other place it is used is on the Consumer, which is exactly the same (just a String). 
While the JSON representation is:
```
"releaseVer": {
    "releaseVer": "string",
    "created": "2017-11-21T16:02:22.861Z",
    "updated": "2017-11-21T16:02:22.861Z"
  }
```
Do these timestamps get used/give us any value? If not it should be just a String everywhere then.

About this specific change though, not sure what you mean by not revert this code.
Should I just do the validation against the model object's release instead of the dto object's one because it's simpler?

Reply:
The created/updated fields are remnants of our backend leaking -- consumers don't actually care about it. Also, we don't want to unnecessarily abbreviate our API calls.

What I meant by "revert this code" is by keeping what is on this side of the diff rather than the new stuff.

Reply:
Sorry, but I'm still unsure which parts I should revert. Should I revert all the changes to the whole endpoint (which used ActivationKeyData instead of ActivationKeyDTO)? Or just revert the 2nd null check in the if block?

Reply:
Keep the method names using the full name: ReleaseVersion, only manage a single string value and let a JSON serializer/deserializer handle the inflating part. In fact, you can probably just end up using the same serializer that you use for the product stuff for this as well.